antismash/common/secmet/locations.py

```diff
@@ -30,45 +30,46 @@ def convert_protein_position_to_dna(start: int, end: int, location: FeatureLocat
     """
     if not 0 <= start < end <= len(location) // 3:
         raise ValueError("Protein positions %d and %d must be contained by %s" % (start, end, location))
-    if location.strand == -1:
-        dna_start = location.start + len(location) - end * 3
-        dna_end = location.start + len(location) - start * 3
-    else:
-        dna_start = location.start + start * 3
-        dna_end = location.start + end * 3
-
-    # only CompoundLocations are complicated
-    if not isinstance(location, CompoundLocation):
-        if not location.start <= dna_start < dna_end <= location.end:
-            raise ValueError(("Converted coordinates %d..%d "
-                              "out of bounds for location %s") % (dna_start, dna_end, location))
-        return dna_start, dna_end
+    dna_start = start * 3
+    dna_end = end * 3
 
-    parts = sorted(location.parts, key=lambda x: x.start)
-    gap = 0
-    last_end = parts[0].start
+    processed = 0
```

Comment:
Having been reminded of the existence of frameshifts causing overlapping location parts [here](https://github.com/antismash/antismash/pull/48/commits/74b18dd89047887e3cba68979e0416b672682c37), I don't think this method is going to work. Using the lengths of parts in this way is going to cause some problems because the counts ignore that the overlap is there.

GCF_000265205.3 is an example assembly that contains this kind of location.

Reply:
exactly what i meant by adding the todo: in line 71 in the first of the two commits. Such a combined subroutine would only work on locations, processed = 0 will still exist there, but the locations will be returned, not dna_start, dna_end.